src/app/inbound/mod.rs

```diff
@@ -31,6 +35,7 @@ pub fn server<P>(
     endpoint_http_metrics: super::HttpEndpointMetricsRegistry,
     route_http_metrics: super::HttpRouteMetricsRegistry,
     transport_metrics: transport::metrics::Registry,
+    span_sink: mpsc::Sender<oc::Span>,
```

Comment:
should this be a SpanSink?

Reply:
What happens when tracing is disabled?

Reply:
This should not be a `SpanSink`.  That trait is for `trace_context::Span` whereas this channel is of `oc::Span` (i.e. the protobuf object).